scripts/generate_verify_psa_macros.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python3
+
+"""Generate verify_psa_macros.c"""
+
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import re
+
+OUTPUT_TEMPLATE = '''\
+/* Automatically generated by generate_verify_psa_macros.py. DO NOT EDIT. */
+
+#include <stdio.h>
+
+#include "psa/crypto.h"
+
+{header}
+
+#define hash_alg   PSA_ALG_SHA_256
+#define mac_alg    PSA_ALG_CMAC
+#define ka_alg     PSA_ALG_ECDH
+#define kdf_alg    PSA_ALG_HKDF(PSA_ALG_SHA_256)
+#define mac_length 4
+#define group      0x2A
+#define curve      0x2A
+#define type       PSA_KEY_TYPE_RSA_PUBLIC_KEY
+
+int main() {{
+    int ret = 0;
+
+    {tests}
+
+    return ret;
+}}
+'''
+
+HEADER_TEMPLATE = '''\
+#ifdef {name}
+#define SPEC_{name}{parameters} {expansion}
+#else
+#define {name}{parameters} {expansion}
+#endif
+'''
+
+TEST_TEMPLATE = '''\
+#ifdef SPEC_{name}
+    if ({name}{parameters} != SPEC_{name}{parameters}) {{
+        fprintf(stderr, "ERROR: %s doesn't match value in spec\\n", "{name}{parameters}");
+        ret = 1;
+    }}
+#else
+    printf("WARN:  %s is not defined\\n", "{name}{parameters}");
+#endif
+'''
+
+TEST_DEFINED_TEMPLATE = '''\
+#ifndef {name}
+    printf("WARN:  %s is not defined\\n", "{name}{parameters}");
+#endif
+'''
+
+class MacroCollector():
```

Comment:
For information, I'm working on a script that generates storage test cases, and it also uses the `MacroCollector` class. So far I've only made a tiny patch and then I derive from it. As discussed internally, I'm going to create a directory `scripts/mbedtls_dev` and start putting non-executable Python modules there.

Reply:
#3953 creates `scripts/mbedtls_dev`. `MacroCollector` could be moved to a new module there. (`import generate_psa_constants.MacroCollector` would work, but creating a new module is cleaner.)